### DISCUSSION TITLE
Persisted File loses most of it's data after refresh

### DISCUSSION THREAD

**Discussion by gurkang** [pmndrs/jotai#1863] [Q&A]
I've got an atom that I want to persist across the application for different usecases. This atom has a field that is a Blob.

```
export const formAtom = atomWithStorage<FormData>(
  "coverLetterFormAtom",
  {
    resumeFile: null,
  }
);
```

FormData type:

```
export type FormData = {
  resumeFile: Blob | null;
};
```

I set the storage using react-dropzone:

```
const [form, setForm] = useAtom(formAtom);
 const onDrop = useCallback((acceptedFiles: any) => {
    if (form) setForm({ ...form, resumeFile: acceptedFiles[0] });
  }, []);
```

This works on initial load and when I drop a new file I can see following data as I inspect the state:
```
resumeFile:{arrayBuffer: ƒ arrayBuffer() {},
path:"example.pdf"
name:"example.pdf"
lastModified:1679650210223
lastModifiedDate:Fri Mar 24 2023 1:3:10 GMT+0700 (Indochina Time)
webkitRelativePath:""
size:41902
type:"application/pdf"
arrayBuffer:ƒ arrayBuffer() {}
slice:ƒ slice() {}
stream:ƒ stream() {}
text:ƒ text() {}
```

However, after refreshing the page and jotai tries to load the same file I get only a small non functioning subset:

```
esumeFile:{path: "example.pdf"}
path:"example.pdf"
```

Why am I "loosing" a bunch of data on the refresh. Is jotai not storing the file correctly or what am I missing when it comes to persisting a file like this across refreshes/state changes?

**Comment by dai-shi** [comment]
By default, atomWithStorage use JSON.parse/stringify for de-/serialization, which doesn't support Blob. You want to provide a custom storage with de-/serialization that supports Blob. Not sure how easy it is. Maybe there's some technique or library that helps serializing Blob.
